html/modules/custom/hr_paragraphs/hr_paragraphs.module

```diff
@@ -627,6 +640,161 @@ function hr_paragraphs_preprocess_paragraph__group_pages(&$variables) {
   ];
 }
 
+/**
+ * Implements hook_preprocess_paragraph__type().
+ */
+function hr_paragraphs_preprocess_paragraph__reliefweb_river(&$variables) {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  // Exclude special pages.
+  if (!$paragraph->hasField('field_reliefweb_url') || $paragraph->field_reliefweb_url->isEmpty()) {
+    return;
+  }
+
+  $url = $paragraph->field_reliefweb_url->first()->uri;
+  /** @var \Drupal\hr_paragraphs\Controller\ReliefwebController */
+  $reliefweb_controller = \Drupal::service('hr_paragraphs.reliefweb_controller');
+  $conditions = $reliefweb_controller->parseReliefwebUrl($url);
+  $parameters = $reliefweb_controller->buildReliefwebParameters(0, 10, []);
+
+  // Remove facets.
+  unset($parameters['facets']);
+
+  // Add conditions.
+  if ($reliefweb_controller->reliefwebQueryType($url) === 'advanced-search') {
+    foreach ($conditions as $condition) {
+      $parameters['filter']['conditions'][] = [
+        'field' => $condition['field'],
+        'value' => $condition['value'],
+      ];
+    }
+  }
+  else {
+    unset($parameters['filter']);
+    $parameters['query'] = [
+      'value' => $conditions[0],
+      'operator' => 'AND',
+    ];
+  }
+
+  $results = $reliefweb_controller->executeReliefwebQuery($parameters);
+
+  $variables['content']['rw_list'] = [
+    '#theme' => 'hr_paragraphs_rw_list',
+    '#data' => $reliefweb_controller->buildReliefwebObjects($results),
+    '#total' => 10,
+  ];
+}
+
+/**
+ * Implements hook_preprocess_paragraph__type().
+ */
+function hr_paragraphs_preprocess_paragraph__reliefweb_document(&$variables) {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  // Exclude special pages.
+  if (!$paragraph->hasField('field_reliefweb_url') || $paragraph->field_reliefweb_url->isEmpty()) {
+    return;
+  }
+
+  $url = $paragraph->field_reliefweb_url->first()->uri;
+  $flags = LIBXML_NONET | LIBXML_NOBLANKS | LIBXML_NOERROR | LIBXML_NOWARNING;
+
+  $data = file_get_contents($url);
+  $dom = new \DOMDocument();
+  $dom->loadHTML($data, $flags);
+
+  $nid = FALSE;
+  $nodes = $dom->getElementsByTagName('link');
+  foreach ($nodes as $node) {
+    if ($node->getAttribute('rel') === 'shortlink') {
+      $nid = str_replace('https://reliefweb.int/node/', '', $node->getAttribute('href'));
+      break;
+    }
+  }
+
+  if (!$nid) {
+    return;
+  }
+
+  /** @var \Drupal\hr_paragraphs\Controller\ReliefwebController */
+  $reliefweb_controller = \Drupal::service('hr_paragraphs.reliefweb_controller');
+  $parameters = $reliefweb_controller->buildReliefwebParameters(0, 1, []);
+
+  // Remove facets.
+  unset($parameters['facets']);
+
+  // Add conditions.
+  $parameters['filter']['conditions'][] = [
+    'field' => 'id',
+    'value' => $nid,
+  ];
```

Comment:
@rupl @attiks For info, if you have the URL of a RW document, you should be able get the corresponding API entry with the following filter:

```
 $parameters['filter']['conditions'][] = [
     'field' => 'url_alias',
     'value' => $url,
   ];
``` 

Hopefully, that would avoid having to fetch the page to extract the shortlink.

Reply:
Nice, fixed in https://github.com/UN-OCHA/hrinfo-site-2022/pull/38/commits/1fb30d6158670abf7326f824f4c46bc931b6de35